timelock-impl/src/main/java/com/palantir/atlasdb/timelock/management/AllNodesDisabledNamespacesUpdater.java

```diff
@@ -0,0 +1,271 @@
+import com.palantir.atlasdb.timelock.api.Namespace;
+import com.palantir.atlasdb.timelock.api.ReenableNamespacesRequest;
+import com.palantir.atlasdb.timelock.api.ReenableNamespacesResponse;
+import com.palantir.atlasdb.timelock.api.SingleNodeUpdateResponse;
+import com.palantir.atlasdb.timelock.api.SuccessfulDisableNamespacesResponse;
+import com.palantir.atlasdb.timelock.api.SuccessfulReenableNamespacesResponse;
+import com.palantir.common.concurrent.CheckedRejectionExecutorService;
+import com.palantir.common.streams.KeyedStream;
+import com.palantir.logsafe.SafeArg;
+import com.palantir.logsafe.logger.SafeLogger;
+import com.palantir.logsafe.logger.SafeLoggerFactory;
+import com.palantir.paxos.BooleanPaxosResponse;
+import com.palantir.paxos.PaxosQuorumChecker;
+import com.palantir.paxos.PaxosResponse;
+import com.palantir.paxos.PaxosResponsesWithRemote;
+import com.palantir.paxos.WrappedPaxosResponse;
+import com.palantir.tokens.auth.AuthHeader;
+import java.time.Duration;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+public class AllNodesDisabledNamespacesUpdater {
+    private static final SafeLogger log = SafeLoggerFactory.get(AllNodesDisabledNamespacesUpdater.class);
+
+    private final AuthHeader authHeader;
+    private final ImmutableList<DisabledNamespacesUpdaterService> remoteUpdaters;
+    private final Map<DisabledNamespacesUpdaterService, CheckedRejectionExecutorService> remoteExecutors;
+    private final TimelockNamespaces localUpdater;
+    private final Supplier<UUID> lockIdSupplier;
+    private final int clusterSize;
+
+    @VisibleForTesting
+    AllNodesDisabledNamespacesUpdater(
+            AuthHeader authHeader,
+            ImmutableList<DisabledNamespacesUpdaterService> remoteUpdaters,
+            Map<DisabledNamespacesUpdaterService, CheckedRejectionExecutorService> remoteExecutors,
+            TimelockNamespaces localUpdater,
+            Supplier<UUID> lockIdSupplier) {
+        this.authHeader = authHeader;
+        this.remoteUpdaters = remoteUpdaters;
+        this.remoteExecutors = remoteExecutors;
+        this.localUpdater = localUpdater;
+        this.lockIdSupplier = lockIdSupplier;
+        this.clusterSize = remoteUpdaters.size() + 1;
+    }
+
+    public static AllNodesDisabledNamespacesUpdater create(
+            AuthHeader authHeader,
+            ImmutableList<DisabledNamespacesUpdaterService> updaters,
+            Map<DisabledNamespacesUpdaterService, CheckedRejectionExecutorService> executors,
+            TimelockNamespaces localUpdater) {
+        return new AllNodesDisabledNamespacesUpdater(authHeader, updaters, executors, localUpdater, UUID::randomUUID);
+    }
+
+    public DisableNamespacesResponse disableOnAllNodes(Set<Namespace> namespaces) {
```

Comment:
Let's add some javadocs that explain a bit what this does and the guarantees / pitfalls